internal/engine/build_and_deployer.go

```diff
@@ -52,6 +55,26 @@ func NewCompositeBuildAndDeployer(builders BuildOrder, shouldFallBack FallbackTe
 
 func (composite *CompositeBuildAndDeployer) BuildAndDeploy(ctx context.Context, manifest model.Manifest, currentState BuildState) (BuildResult, error) {
 	var lastErr error
+	changedConfigFiles := getChangedConfigFiles(currentState.FilesChanged(), manifest)
+	if len(changedConfigFiles) > 0 {
+		logger.Get(ctx).Verbosef("Detected a config file change (%v), re-executing tiltfile and doing an image build", changedConfigFiles)
+		lastBuilder := composite.builders[len(composite.builders)-1]
```

Comment:
This is intentionally hacky. In the future we'll be smarter about what kind of build we do depending on what has changed.